CodeEditor/Sources/CodeEditor/UXCodeTextView.swift

```diff
@@ -338,6 +336,84 @@ final class UXCodeTextView: UXTextView, HighlightDelegate, UIScrollViewDelegate
         }
         UIGraphicsPopContext()
     }
+    
+    func updateLightBulbs() {
+        // delete and add new lightbulbs when code changes
+        for lightBulb in lightBulbs {
+            lightBulb.removeFromSuperview()
+        }
+        lightBulbs = []
+        addLightbulbs()
+    }
+    
+    // calculates the offset of the given line if line wrapping occurs
+    private func calculateWrapOffsetOf(_ lineNumber: Int) -> Int {
+        var offset = 0, curr = 1, subParaCount = 0
+        var subParaRange = NSRange(location: 0, length: 0)
+        layoutManager.enumerateLineFragments(forGlyphRange: layoutManager.glyphRange(for: textContainer)) { _, _, _, glyphRange, stop in
+            if curr == lineNumber {
+                let charRange = self.layoutManager.characterRange(forGlyphRange: glyphRange, actualGlyphRange: nil)
+                let paraRange = NSString(string: self.textStorage.string).paragraphRange(for: charRange)
+                if subParaRange != paraRange {
+                    // if done during subPara jump back to actual line
+                    offset -= subParaCount
+                }
+                stop.pointee = true
+            } else {
+                let charRange = self.layoutManager.characterRange(forGlyphRange: glyphRange, actualGlyphRange: nil)
+                if subParaRange.length > 0 {
+                    subParaRange.length += charRange.length
+                } else {
+                    subParaRange = charRange
+                }
+                let paraRange = NSString(string: self.textStorage.string).paragraphRange(for: charRange)
+                if subParaRange != paraRange {
+                    offset += 1
+                    subParaCount += 1
+                } else {
+                    subParaRange = NSRange(location: 0, length: 0)
+                    subParaCount = 0
+                }
+                curr += 1
+            }
+        }
+        return offset
+    }
+    
+    func addLightbulbs() {
+        var lineNumber = 1
+        layoutManager.enumerateLineFragments(forGlyphRange: layoutManager.glyphRange(for: textContainer)) { rect, _, _, _, _ in
+            let offset = self.calculateWrapOffsetOf(lineNumber)
+            if let feedback = self.feedbackSuggestions.first(where: { $0.fromLine == lineNumber - offset }) {
+                self.currFeedbackId = feedback.id.uuidString
```

Comment:
This should be set when the button is pressed, not when it is created. Currently, there is a bug where with 2 feedback suggestions in one, the same one is displayed for both:

https://user-images.githubusercontent.com/9006596/215584509-80bc6009-a7be-4d42-9355-7d99b1a0135e.mp4

Alternatively (even better IMO), this button click listener could just pass the feedback ID to `onLightbulbTap` directly instead of temporarily storing it in an attribute.